spec/models/claim/claims_preventing_payment_finder_spec.rb

```diff
@@ -0,0 +1,86 @@
+require "rails_helper"
+
+RSpec.describe Claim::ClaimsPreventingPaymentFinder do
+  subject(:finder) { described_class.new(claim) }
+
+  describe "#claims_preventing_payment" do
+    let(:personal_details) do
+      {
+        national_insurance_number: generate(:national_insurance_number),
+        bank_account_number: "32828838",
+        bank_sort_code: "183828",
+        first_name: "Boris",
+      }
+    end
+    let(:claim) { create(:claim, :submitted, personal_details) }
+    subject(:claims_preventing_payment) { finder.claims_preventing_payment }
+
+    context "when there is another claim with the same National Insurance number, with inconsistent personal details that would prevent us from running payroll" do
+      let(:inconsistent_personal_details) do
+        personal_details.merge(
+          bank_account_number: "87282828",
+          bank_sort_code: "388183",
+        )
+      end
+
+      context "when the other claim is not yet approved" do
```

Comment:
The embedded contexts here are a bit tricky to follow, do you reckon you could flatten these, so they look a bit like this?

```ruby
it "does not include the other claim when the other claim is not yet approved" do
  other_claim = create(:claim, :submitted, inconsistent_personal_details)
  expect(claims_preventing_payment).to be_empty
end
```

Reply:
Sure thing - changed.